### PR TITLE
Tidy up the docker-compose file a bit

### PR DESCRIPTION
**Pull request by SammyK**
### Description

Not all of the active containers were represented in the main docker-compose file so I added them. I also grouped them together to tidy things up a bit.

I don't use any of the images past the `--- Legacy Images ---` comment. If no one else uses them, I can update this PR to delete them.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -108,19 +124,12 @@ services:
...
-  '7.3-centos6': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-7.3_centos-6' }
-  '7.3-buster': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-7.3_buster' }
   '7.3-debug': { <<: *base_php_service, image: 'datadog/docker-library:ddtrace_alpine_php-7.3-debug' }
   '7.3-zts-debug': { <<: *base_php_service, image: 'datadog/docker-library:ddtrace_alpine_php-7.3-zts-debug' }
   '7.3-zts': { <<: *base_php_service, image: 'datadog/docker-library:ddtrace_alpine_php-7.3-zts' }
-  '7.4-buster': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-7.4_buster' }
   '7.4-debug': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-7.4-debug-alpine-3.11' }
   '7.4-debug-asan': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-7.4-debug-asan-buster' }
-  '8.0-buster': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-8.0_buster' }
   '8.0-shared-ext': { <<: *base_php_service, image: 'datadog/dd-trace-ci:php-8.0-shared-ext' }
```
Comment (labbati) [inline docker-compose.yml:132]: If we go on with removing the legacy images, I think we still need this `8.0-shared-ext` correct?

**Review by labbati — CHANGES_REQUESTED** [review]
Thanks @SammyK, I was about to do the same. I am 👍🏻  with removing the legacy ones but I added a comment to `8.0-shared-ext`.

**Review by labbati — APPROVED** [review]
Love it. Thanks @SammyK

File: docker-compose.yml (hunk 1 above)
Comment (SammyK) [reply docker-compose.yml]: Nice catch! I've moved shared-ext to the Buster section and removed some of the legacy references and some old Dockerfiles.

**Comment by SammyK** [comment]
That was fast! Thanks @labbati. :)

**Pull request merged**